### PR TITLE
Fix file preview paths

### PR DESCRIPTION
**Pull request by ikisler**
When the wiki does not sit in the root directory for the server, the file paths are incorrect for the file preview function.  Added code so that it now grabs the current directory in the file path.

### CODE REVIEW

File: modules/feature-upload.php [hunk 1]
```diff
@@ -291,7 +304,7 @@
 				exit();
 			}
 			
-			$filepath = $env->storage_prefix . $pageindex->{$env->page}->uploadedfilepath;
+			$filepath = getcwd() . "/" . $env->storage_prefix . $pageindex->{$env->page}->uploadedfilepath;
```
Comment (sbrl) [inline modules/feature-upload.php]: What if the storage prefix is absolute? Just to be a pain :P

**Comment by sbrl** [comment]
I've just tidied up an error message a bit :smiley_cat:

File: modules/feature-upload.php (hunk 1 above)
Comment (ikisler) [inline modules/feature-upload.php]: Too true.  Maybe we should check to see if the dir exists as-is, call getcwd() if it doesn't, then check to see if _that_ exists, and if it still isn't right, throw an error?

Comment (ikisler) [inline modules/feature-upload.php]: Hrm... the more I mess with this, the more I think that the problem is different than I had thought.  When I use the original code for the filepath (without the `getcwd`), but put it through `realpath`, it has the correct value.  I found that some people had problems with imagick not working with relative paths, but all the bug reports I can find are related to PHP 5.4, not 7.  I assume that the original works fine for you?

Comment (sbrl) [inline modules/feature-upload.php]: Yeah, imagick appears to take relative paths just fine IIRC.

And reading this again with fresh eyes, the `storage_directory` is meant to be a path specified by the user that should point to the directory in which the wiki's data is actually stored. e.g. if `index.php` is stored in `/srv/www/sekretapplewiki`, but the data is actually stored in `/srv/data/applestuff`, then `storage_prefix` could either be `/srv/data/applestuff` or `../../data/applestuff`. IIRC it defaults to `.`.

This does rely on the current directory being the directory in which `index.php` is stored, which I'm not sure how often that is the case (it has been for me).

Thanks for your effort! It's been quite enjoyable seeing someone else who's as interested in this project as I am :smiley_cat:

Comment (ikisler) [inline modules/feature-upload.php]: Sorry this took so long to get back to -- work has been nuts this past week.

I tested this issue on the two other environments I have available -- a linux server and another Windows machine, and could not reproduce the issue with either of them.  I can't seem to track down the issue, but passing it to imagick as an absolute path solves the problem for me.  I went ahead an made a commit wrapping the path in a realpath() call.

Thank you for being so open to contributions! =D  I've enjoyed working with this codebase.

Comment (sbrl) [inline modules/feature-upload.php]: Ah, no worries! I know the feeling... :P

Hrm. Well, we can't have some environments working and others not working, notw can we? :P Yeah, I think I've heard something about canonical pathnames and imagick somewhere before. This looks ok, so I'll merge :smiley_cat:

**Pull request merged**